include/affinity.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <assert.h>
+
+typedef uint64_t uint64;
+typedef uint32_t uint32;
+
+#define MAX_CORES ((uint64)2048)
+#define AFFINITY_MASK_BITS_PER_QUAD ((uint64)64)
+#define AFFINTY_MASK_NUM_QUADS (MAX_CORES / AFFINITY_MASK_BITS_PER_QUAD)
+
+struct _affinity_mask {
```

Comment:
This (and most of the contents of this header) are really more of an implementation detail and should probably be hidden.